crates/runtime/src/instance.rs

```diff
@@ -855,6 +858,125 @@ impl Instance {
         }
     }
 
+    /// Initialize the VMContext data associated with this Instance.
+    ///
+    /// The `VMContext` memory is assumed to be uninitialized; any field
+    /// that we need in a certain state will be explicitly written by this
+    /// function.
+    unsafe fn initialize_vmctx(&mut self, store: StorePtr, imports: Imports) {
+        let module = self.module().clone();
```

Comment:
To avoid this arc clone/decrement could the module be passed in? (since I think callers always have it available)

Reply:
I'll do that and assert the two modules match.